services/can_logger/can_logger.py

```diff
@@ -8,51 +7,124 @@
 import sys
 import collections
 import multiprocessing as mp
+import logging
 from datetime import datetime
 from psycopg2 import OperationalError
 from psycopg2.extras import execute_values
 
-
 def write_to_csv(wr_buff, can_bus):
-
     # Write to CSV from write buffer. First item is timestamp, second is
-    # can_id and third is can_data
-
-    with open(f'/data/log/{can_bus}.csv', mode='a') as logfd:
-
-        logcsv = csv.writer(logfd, delimiter=',', quotechar='"',
-                            quoting=csv.QUOTE_MINIMAL)
-
-        logcsv.writerows(wr_buff)
+    # can_bus, third is can_id, and last is can_data
+    # Generate timestamp to rotate the log
+    global log_path
+    #timestamp = rotate_log()
+    while(not log_path):
+        try:
+            with open(f'/data/log/can/{can_bus}.csv', mode='a') as logfd:
+                logcsv = csv.writer(logfd, delimiter=',', quotechar='"',
+                                    quoting=csv.QUOTE_MINIMAL)
+                logcsv.writerows(wr_buff)
+                log_path = True
+        except(FileNotFoundError):
+            path = '/data/log/can'
+            os.mkdir(path)
+            print(f'Logging path set to {path}')
 
     print(f'{can_bus}: Wrote successfully to CSV')
```

Comment:
This should be debug, I assume this creates a lot of stdout that ends up written to the disk by docker.

Reply:
Indeed. However, this is printed once at most. Even in the scenario that the container is continually failing, the path will be created the first time the program runs and handles the exception. After that, the program would not raise that exception anymore.

Reply:
I think this is referring to line 33

Reply:
My bad. I though it referred to the green highlighted code. Indeed, this will be removed.

Reply:
Good point. I don't understand why they are green?